src/drivers/intel/cavs/ipc.c

```diff
@@ -185,6 +194,15 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	hdr = mailbox_validate();
 #endif
 
+#ifdef IPC4_EARLY_ACK
+	/*
+	 * notify host that we have received the message and
+	 * do this before we send the response
+	 */
+	ipc_write(IPC_DIPCTDR, ipc_read(IPC_DIPCTDR) | IPC_DIPCTDR_BUSY);
+	ipc_write(IPC_DIPCTDA, ipc_read(IPC_DIPCTDA) | IPC_DIPCTDA_DONE);
+#endif
+
```

Comment:
It is a test. I prefer to do it in ipc4/handler.c without ifdef.

Reply:
@RanderWang  Not sure I follow. This is writing to IPC hw registers and it thus should be done in the driver, not the protocol code in ipc4/handler.c. Right? The ifdef is needed as same driver is used for IPC3 and there the semantics are different ACK/REPLY are the same thing, so the proper place to do is later.